Content.Server/Shuttles/Systems/ArrivalsSystem.cs

```diff
@@ -164,13 +171,22 @@ private void OnArrivalsFTL(EntityUid shuttleUid, ArrivalsShuttleComponent compon
             DumpChildren(shuttleUid, ref args, pendingEntQuery, arrivalsBlacklistQuery, mobQuery, xformQuery);
         }
 
+        // if (!TryComp<StationDataComponent>(component.Station, out var data))
+        //     return;
+        //
+        // var targetGrid = _station.GetLargestGrid(data);
         var pendingQuery = AllEntityQuery<PendingClockInComponent, TransformComponent>();
 
         // Clock them in when they FTL
         while (pendingQuery.MoveNext(out var pUid, out _, out var xform))
         {
             // Cheaper to iterate pending arrivals than all children
-            if (xform.GridUid != shuttleUid)
+            if (xform.GridUid == shuttleUid)
+            {
+                // Teleport them on to the station.
+                TeleportToMapSpawn(pUid, component.Station, xform);
+            }
+            if (xform.MapUid == arrivalsXform.MapUid)
```

Comment:
i don't quite understand why the grid check was changed. It also probably makes more sense to wrap this in an "if-else" considering that getting teleported to map spawn should always remove you from the pending arrivals.

Reply:
Perhaps needs better comments. 

Before: If you were on the shuttle when it goes to warp, you either lose the ability to go to warp in shuttle next time or get booted off (regardless of shuttle starting point).

Now: 
1. If shuttle is heading away from arrivals, do nothing (fixes arrivals stranding bugs also)
2. On shuttle? Warp to a spawn point
3. On arrivals? Keep your "warp" coupon
4. Not on arrivals? Remove your coupon.